core/src/main/resources/dev/process_supported_files.py

```diff
@@ -0,0 +1,237 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+"""
+Description:
+    This Python script goes over all directories of different spark versions under the
+    spark-rapids/tools/generated_files folder, contructs unions of the supported CSV
+    files (supportedDataSource.csv, supportedExecs.csv, and supportedExprs.csv) and
+    writes the results to new CSV files.
+
+Dependencies:
+    - numpy >= 1.23.3
+    - pandas >= 2.0.3
+
+Usage:
+    python process_supported_files.py generated_files_path [--configs configs_file] [--output output_directory]
+"""
+
+
+import argparse
+import json
+import logging
+import os
+from enum import IntEnum
+
+import pandas as pd
+
+
+class SupportLevel(IntEnum):
+    """
+    IntEnum Class for support level types to facilitate easy comparison.
+
+    Enum members:
+    - NA: Not Applicable.
+    - NS: Not Supported.
+    - CO: Configured Off.
+    - PS: Partially Supported.
+    - S: Supported.
+    """
+    NA = 1
+    NS = 2
+    CO = 3
+    PS = 4
+    S = 5
```

Comment:
It is difficult to know why a given operator is getting its level of support; especially that we want to override that support on-demand. For instance, when Hao wants to troubleshoot an operator, he won't be able to tell whether the value comes from the plugin,  overridden by the tools, or coming from a specific version of Spark.

I suggest we define two other members:

- `TOFF` standing for "tools-OFF".
- `TON` standing for "tools-ON".

This will make it easy for anyone to know that this operator is turned-on/off on the tools side.

- when we want to enforce disabling an operator then:
  - Add an entry in the override-config file defining its support level to be "TOFF".
  - Run the script to generate new CSV file which will have TOFF
- when we want to enforce enabling an operator then:
  - Add an entry in the override-config file defining its support level to be "TON".
  - Run the script to generate new CSV file which will have TON

This change requires that we mirror those two members on the Scala side as well.